apps/emqx/src/config/emqx_config_logger.erl

```diff
@@ -276,13 +277,28 @@ do_formatter(text, CharsLimit, SingleLine, TimeOffSet, Depth, TsFormat, WithMfa,
 log_filter(?AUDIT_HANDLER, _Conf) ->
     [{filter_audit, {fun ?MODULE:filter_audit/2, stop}}];
 log_filter(_, Conf) ->
+    Filters = [
+        {drop_noproc_shutdown_errors, {fun ?MODULE:is_supervisor_noproc_shutdown_error/2, stop}}
+    ],
     case conf_get("supervisor_reports", Conf) of
-        error -> [{drop_progress_reports, {fun logger_filters:progress/2, stop}}];
-        progress -> []
+        error -> [{drop_progress_reports, {fun logger_filters:progress/2, stop}} | Filters];
+        progress -> Filters
     end.
 
 filter_audit(_, _) -> stop.
 
+is_supervisor_noproc_shutdown_error(
+    #{msg := {report, #{label := {supervisor, shutdown_error}, report := Report}}}, OnMatch
+) when is_list(Report) ->
+    case proplists:get_value(reason, Report) of
+        noproc ->
```

Comment:
My 2 cents. Those don't feel like events that are okay to ignore, it hints that something weird is going on in the system that certainly needs attention.

Reply:
It could be just a race condition, as it seems to be the case for resource manager: i) a call is made to the process to [stop it](https://github.com/emqx/emqx/blob/464e2027421c10009a0a2fff219bd0303e466dce/apps/emqx_resource/src/emqx_resource_manager.erl#L278-L296); ii) the call times out; iii) the `after` clause ends up calling `supervisor:terminate_child`; iv) while the supervisor is finding the child to terminate, the process actually finishes terminating, and the supervisor receives a [`noproc` reason](https://github.com/emqx/otp/blob/e6c6b1a4fffffe77fe36da1e38390b856cfa3e6e/lib/stdlib/src/supervisor.erl#L913) back.

Yet, this is just one particular case: I don't think there's a way to distinguish between a harmless race and a bug, indeed.  Maybe the user should just ignore such log noise.  :thinking:

Reply:
Also, I believe the user in question is using a version that doesn't contain the `force_kill` patch, so any timeouts while stopping a resource might lead to such noise.

I've added a monitor to the process and now wait for the exit signal in force_kill.  This way, we shouldn't get the race condition of the process dying while the supervisor selects the child pid.